signature/src/keypair.rs

```diff
@@ -0,0 +1,188 @@
+    pub fn new() -> Self {
+        let sk = ed25519_zebra::SigningKey::new(rand::thread_rng());
+        let vk = ed25519_zebra::VerificationKey::from(&sk);
+
+        return Self {
+            private_key: sk,
+            public_key: vk,
+            relation: crate::suite::VerificationRelation::AssertionMethod,
+
+            master_public_key: vk,
+            master_private_key: sk,
+
+            authetication_public_key: vk,
+            authetication_private_key: sk,
+
+            capability_invocation_public_key: vk,
+            capability_invocation_private_key: sk,
+
+            capability_delegation_public_key: vk,
+            capability_delegation_private_key: sk,
+
+            assertion_method_public_key: vk,
+            assertion_method_private_key: sk,
+        };
+    }
+
+    pub fn get_public_key_encoded(&self) -> String {
+        multibase::encode(multibase::Base::Base58Btc, self.public_key)
+    }
+
+    pub fn get_master_public_key_encoded(&self) -> String {
+        multibase::encode(multibase::Base::Base58Btc, self.master_public_key)
+    }
+
+    pub fn get_verification_method(self, relation: crate::suite::VerificationRelation) -> String {
+        match relation {
+            crate::suite::VerificationRelation::AssertionMethod => {
+                format!(
+                    "{}{}#{}",
+                    DID_PREFIX.to_string(),
+                    multibase::encode(multibase::Base::Base58Btc, self.master_public_key),
+                    multibase::encode(multibase::Base::Base58Btc, self.assertion_method_public_key)
+                )
+            }
+            crate::suite::VerificationRelation::Authentication => {
+                format!(
+                    "{}{}#{}",
+                    DID_PREFIX.to_string(),
+                    multibase::encode(multibase::Base::Base58Btc, self.master_public_key),
+                    multibase::encode(multibase::Base::Base58Btc, self.authetication_public_key)
+                )
+            }
+            crate::suite::VerificationRelation::CapabilityInvocation => {
+                format!(
+                    "{}{}#{}",
+                    DID_PREFIX.to_string(),
+                    multibase::encode(multibase::Base::Base58Btc, self.master_public_key),
+                    multibase::encode(
+                        multibase::Base::Base58Btc,
+                        self.capability_invocation_public_key
+                    )
+                )
+            }
+            crate::suite::VerificationRelation::CapabilityDelegation => {
+                format!(
+                    "{}{}#{}",
+                    DID_PREFIX,
+                    multibase::encode(multibase::Base::Base58Btc, self.master_public_key),
+                    multibase::encode(
+                        multibase::Base::Base58Btc,
+                        self.capability_delegation_public_key
+                    )
+                )
+            }
+        }
+    }
```

Comment:
This can be simplified by assigning the match arm to a value:
```suggestion
pub fn get_verification_method(self, relation: crate::suite::VerificationRelation) -> String {
        let verification_key = match relation {
            crate::suite::VerificationRelation::AssertionMethod => self.assertion_method_public_key,
            crate::suite::VerificationRelation::Authentication => self.authetication_public_key,
            crate::suite::VerificationRelation::CapabilityInvocation => {
                self.capability_invocation_public_key
            }
            crate::suite::VerificationRelation::CapabilityDelegation => {
                self.capability_delegation_public_key
            }
        };
        format!(
            "{DID_PREFIX}{}#{}",
            multibase::encode(multibase::Base::Base58Btc, self.master_public_key),
            multibase::encode(multibase::Base::Base58Btc, verification_key)
        )
    }
```